Add [AV]vip plugin to AVsitter distribution

chotaire opened the pull request:
# AVsitter VIP

## Introduction

Limit access to AVsitter menus to a list of legacy usernames read from notecard

- Compile [AV]vip and place in same prim with other [AV]scripts
- Edit access file and add legacy usernames as desired and copy into same prim
- Inspect AVpos to see an implementation example explained below.

## AVpos Notes

**BUTTON Hidden\*|90410**

This displays the custom button 'Hidden*', using channel 90410 for communications, do not change this number. You can change the button label 'Hidden' to something else, but do not remove the * asterisk behind the name. Remember, do not use 'TOMENU' for this restricted menu as it would override.

**MENU Restricted** 

This is where you put your access restricted entries. You can keep the name 'Restricted' (it will not be visible to the user) or change it to something else as long as it is NOT the same label as BUTTON. If both BUTTON and MENU have the same label, this script will not work.

chotaire committed:
Add [AV]vip plugin to master

Sei-Lisa commented on AVsitter2/Makefile:
```diff
@@ -56,7 +56,8 @@ OPTIMIZED=[AV]sitA.lslo\
  Utilities/AVpos-shifter.lsl\
- Utilities/Noob-detector.lsl
+ Utilities/Noob-detector.lsl\
+ Plugins/AVvip/[AV]vip.lsl
```
This line should be with the other plugins, two lines above.

chotaire replied:
Fixed in https://github.com/AVsitter/AVsitter/pull/115/commits/b9862850ba9df1a74ac1385c614ab19a5bf4169d

Sei-Lisa commented on AVsitter2/Plugins/AVvip/AVpos:
```diff
@@ -0,0 +1,51 @@
+MTYPE 0
```
Examples and documentation should be submitted to the https://github.com/avsitter/avsitter.github.io repository. Their place is not the plugin folder.

chotaire replied:
Documentation removed from plugin directory in https://github.com/AVsitter/AVsitter/pull/115/commits/ffdd1ce6e06c5190f60414e9f80efa1d2dd58453

Sei-Lisa commented on AVsitter2/Plugins/AVvip/[AV]vip.lsl:
```diff
@@ -0,0 +1,98 @@
+            
+            if(id == llGetOwner() || (~idx)) 
+            {
+                llMessageLinked(LINK_SET, 90005, "Sex", id);
```
"Sex"?

chotaire replied:
Changed in https://github.com/AVsitter/AVsitter/pull/115/commits/c97acd833119291f015103c3f620e087162447aa.

Sei-Lisa commented on AVsitter2/Plugins/AVvip/[AV]vip.lsl:
```diff
@@ -0,0 +1,98 @@
+            if(id == llGetOwner() || (~idx)) 
+            {
+                llMessageLinked(LINK_SET, 90005, "Sex", id);
+                llOwnerSay("DEBUG: Access granted to " + llKey2Name(id) + ".");
```
Please remove debug messages.

chotaire replied:
Fixed in https://github.com/AVsitter/AVsitter/pull/115/commits/c97acd833119291f015103c3f620e087162447aa.

Sei-Lisa commented on AVsitter2/Plugins/AVvip/[AV]vip.lsl:
```diff
@@ -0,0 +1,98 @@
+            else 
+            {
+                llMessageLinked(LINK_SET, 90004, "", id);
+                llOwnerSay("DEBUG: Access denied to " + llKey2Name(id) + ".");
```
Please remove debug messages.

chotaire replied:
Fixed in https://github.com/AVsitter/AVsitter/pull/115/commits/c97acd833119291f015103c3f620e087162447aa.

Sei-Lisa commented on AVsitter2/Plugins/AVvip/README.md:
```diff
@@ -0,0 +1,23 @@
+# AVsitter VIP
```
Examples and documentation should be submitted to the https://github.com/avsitter/avsitter.github.io repository. Their place is not the plugin folder.

chotaire replied:
Documentation removed from plugin directory in https://github.com/AVsitter/AVsitter/pull/115/commits/ffdd1ce6e06c5190f60414e9f80efa1d2dd58453

Sei-Lisa reviewed (commented):
I still have to test this and give it more thought, hopefully next month, but on a first pass I've found some issues.

chotaire committed:
Plugin now placed with other plugins

chotaire committed:
Removed examples/docs from plugin directory

chotaire committed:
Removed debug and changed link messages

chotaire commented:
Fixed all these issues, push rights to fork granted to @Sei-Lisa . Happy further reviewing and thanks for your time.

chotaire committed:
Bump [AV]vip version to v0.1.1

Sei-Lisa commented on AVsitter2/Plugins/AVvip/[AV]vip.lsl:
```diff
@@ -0,0 +1,95 @@
+            
+            if(id == llGetOwner() || (~idx)) 
+            {
+                llMessageLinked(LINK_SET, 90005, "VIP", id);
```
This isn't general, as it allows protecting only 1 menu, and in light of #132, it may be necessary to protect all submenus contained.

For that reason, I suggest using the string parameter instead of a fixed string.

Sei-Lisa reviewed (commented):
Sorry for the delay. We're considering this plugin for the next release, but there are some points that need to be addressed first, detailed below. In addition, the name is not sufficiently clear about the purpose of the plug-in. I suggest \[AV\]submenu-whitelist instead.

If you are currently using it, please see #132 about a security issue with this kind of usage of the hidden menus, which was not supported in previous versions.

chotaire committed:
Fix alignments

chotaire committed:
Change link_message numbers

This number was taken by AVsequence. Sei-Lisa has started a new series of numbers at 90400 for extra plugins. Returning with link_message number 90005 on failed access.

chotaire committed:
Use string parameter instead of hardcoded menuname

Made changes to send string parameter instead of hardcoded menu name so users do not need to modify the script to protect multiple submenus (with different button names). Updated documentation inside script.

chotaire committed:
Bump [AV]vip version to v0.1.2

Bump [AV]vip version to v0.1.2 and updated copyrights.

chotaire committed:
Add reference to AV[vip] on 90410

chotaire committed:
Merge branch 'master' into chotaire-avsitter-vip

chotaire commented:
@Sei-Lisa I remember that TOMENU has to be omitted for this script to even work, because otherwise the TOMENU line would override. So if you have patched AVsitter for menus not being displayed unless there is a TOMENU line, this might actually break this script. I'll check into it later. You still have push rights to the fork, feel free to fiddle as well.

Sei-Lisa commented:
Thanks for your work. I'll review later, I just wanted to anticipate this.

> So if you have patched AVsitter for menus not being displayed unless there is a TOMENU line, this might actually break this script.

Not exactly. I've patched AVsitter so that the *listen* event does not honour a request for a submenu if there's no TOMENU line. The 90005 link message still honours it after the patch, therefore I believe this script will keep working and be more secure.

chotaire committed:
Merge branch 'AVsitter:master' into chotaire-avsitter-vip

chotaire committed:
Hardcode msg string to 'Restricted'

Sei-Lisa suggested a change that would no longer send a hardcoded message string to AVsitter. I understood the idea behind this and happily implemented the change, and then I haven't thoroughly tested it. Now I realized that it broke the functionality of the script entirely. The thing is that MENU and BUTTON may not have the same name otherwise AVsitter will compute the request internally instead of passing it via link message. The restricted menu would always be available to everyone. I have now reverted this change, which will fix this plugin.

chotaire committed:
Updated documentation inside script

Updated documentation inside script reflecting the recent msg string change, telling the creator not to use the same label for BUTTON and MENU.

chotaire committed:
Bump version to v0.1.3

chotaire committed:
Merge branch 'AVsitter:master' into chotaire-avsitter-vip
